indy_node/utils/node_control_utils.py

```diff
@@ -26,6 +34,131 @@ def decode_err_handler(error):
 MAX_DEPS_DEPTH = 6
 
 
+class ShellError(subprocess.CalledProcessError):
+    def __init__(self, *args, exc: subprocess.CalledProcessError = None, **kwargs):
+        if exc:
+            super().__init__(exc.returncode, exc.cmd, output=exc.output, stderr=exc.stderr)
+        else:
+            super().__init__(*args, **kwargs)
+
+    @property
+    def stdout_decoded(self):
+        return (self.stdout.decode(locale.getpreferredencoding(), 'decode_errors')
+                if self.stdout else "")
+
+    @property
+    def stderr_decoded(self):
+        return (self.stdout.decode(locale.getpreferredencoding(), 'decode_errors')
+                if self.stderr else "")
+
+
+# TODO use some library instead of dpkg fpr version routine
+class DebianVersion(PackageVersion):
+    cache = {}  # seems not actually necessary
+    # https://www.debian.org/doc/debian-policy/ch-controlfields.html#version
+    re_version = re.compile(r'(?:([0-9]):)?([0-9][a-zA-Z0-9.+\-~]*)')
+
+    @classmethod
+    def _cmp(cls, v1, v2):
+        if v1 == v2:
+            return 0
+        else:
+            cmd = compose_cmd(['dpkg', '--compare-versions', v1, 'gt', v2])
+            try:
+                NodeControlUtil.run_shell_script(cmd)
+            except ShellError as exc:
+                if exc.stderr:
+                    raise
+                else:
+                    return -1
+            else:
+                return 1
+
+    @classmethod
+    def cmp(cls, v1, v2):
+        key = (v1.full, v2.full)
+        if key not in cls.cache:
+            cls.cache[key] = cls._cmp(*key)
+            cls.cache[key[::-1]] = cls.cache[key] * (-1)
```

Comment:
Why do we need this?

Reply:
to reduce number of shell calls since for now we don't have python library API (and don't implement our custom logic) to perform such comparison gently